src/formatter/Cli.hx

```diff
@@ -46,9 +47,18 @@ class Cli {
 		var help = false;
 		var pipemode = false;
 		var argHandler = hxargs.Args.generate([
-			@doc("File or directory with .hx files to format (multiple allowed)")
+			@doc("File or directory with haxe files to format (multiple allowed)")
 			["-s", "--source"] => function(path:String) paths.push(path),
 
+			@doc("File extension to use, defaults to hx")
+			["-e", "--extension"] => function(fileExtension:String) {
+				#if (haxe_ver >= 4.3)
+				extension = fileExtension?.toString() ?? "hx";
+				#else
+				extension = if (fileExtension != null) fileExtension.toString() else "hx";
+				#end
```

Comment:
there is no need to check for >= 4.3, formatter only compiles with 4.3.x or above. and since it's a standalone tool it doesn't need to be compilable with lower Haxe versions.